File: frontend/src/services/album/albumService.js
import api from "../api";

export const getAlbums = async () => {
  const response = await api.get("albumes/");
  return response.data;
};

export const getAlbum = async (id) => {
  const response = await api.get(`albumes/${id}/`);
  return response.data;
};

export const createAlbum = async (data) => {
  const response = await api.post("albumes/", data);
  return response.data;
};

export const updateAlbum = async (id, data) => {
  const response = await api.put(`albumes/${id}/`, data);
  return response.data;
};

export const deleteAlbum = async (id) => {
  const response = await api.delete(`albumes/${id}/`);
  return response.data;
};

File: flutter_app/lib/services/album_service.dart
import '../models/album.dart';
import 'api_service.dart';

class AlbumService {
  final ApiService _apiService = ApiService();

  Future<List<Album>> getAlbums() async {
    final response = await _apiService.get('albumes/');
    if (response is List) {
      return response.map((json) => Album.fromJson(json)).toList();
    } else if (response is Map && response.containsKey('results')) {
      return (response['results'] as List).map((json) => Album.fromJson(json)).toList();
    }
    return [];
  }

  Future<Album> getAlbum(int id) async {
    final response = await _apiService.get('albumes/$id/');
    return Album.fromJson(response);
  }

  Future<Album> createAlbum(Album album) async {
    final response = await _apiService.post('albumes/', album.toJson());
    return Album.fromJson(response);
  }

  Future<Album> updateAlbum(int id, Album album) async {
    final response = await _apiService.put('albumes/$id/', album.toJson());
    return Album.fromJson(response);
  }

  Future<void> deleteAlbum(int id) async {
    await _apiService.delete('albumes/$id/');
  }
}
